flatten_json.go

```diff
@@ -15,14 +16,14 @@ import (
 // There is an exception, namely strings that contain \-prefixed JSON escapes; since we want to work with the
 // actual UTF-8 bytes, this requires re-writing such strings into memory we have to allocate.
 type flattenJSON struct {
-	event      []byte      // event being processed, treated as immutable
-	eventIndex int         // current byte index into the event
-	fields     []Field     // the under-construction return value of the Flatten method
-	skipping   int         // track whether we're within the scope of a segment that isn't used
-	arrayTrail []ArrayPos  // current array-position cookie crumbs
-	arrayCount int32       // how many arrays we've seen, used in building arrayTrail
-	tracker    NameTracker // knows if a segment is used; if not, no need to process
-	cleanSheet bool        // initially true, don't have to call Reset()
+	event      []byte              // event being processed, treated as immutable
+	eventIndex int                 // current byte index into the event
+	fields     []Field             // the under-construction return value of the Flatten method
+	skipping   int                 // track whether we're within the scope of a segment that isn't used
+	arrayTrail []ArrayPos          // current array-position cookie crumbs
+	arrayCount int32               // how many arrays we've seen, used in building arrayTrail
+	tracker    SegmentsTreeTracker // knows if a segment is used; if not, no need to process
```

Comment:
Not obvious why tracker needs to be in the fj struct.   In flattenJSON you pass the `pathnode` argument.

Reply:
You are correct, it's not needed.. removed.